modules/nginx/src/test/java/org/testcontainers/junit/SimpleNginxTest.java

```diff
@@ -8,52 +8,58 @@
 import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
 
 import java.io.*;
+import java.net.URL;
 import java.net.URLConnection;
 
-import static org.rnorth.visibleassertions.VisibleAssertions.assertTrue;
-import static org.rnorth.visibleassertions.VisibleAssertions.info;
-
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.rnorth.visibleassertions.VisibleAssertions.*;
 
 /**
  * @author richardnorth
  */
 public class SimpleNginxTest {
 
-    private static File contentFolder = new File(System.getProperty("user.home") + "/.tmp-test-container");
+    private static String tmpDirectory = System.getProperty("user.home") + "/.tmp-test-container";
 
+    // creatingContainer {
     @Rule
     public NginxContainer nginx = new NginxContainer<>()
-        .withCustomContent(contentFolder.toString())
+        .withCustomContent(tmpDirectory)
         .waitingFor(new HttpWaitStrategy());
+    // }
 
     @SuppressWarnings({"Duplicates", "ResultOfMethodCallIgnored"})
     @BeforeClass
-    public static void setupContent() throws FileNotFoundException {
+    public static void setupContent() throws Exception {
+        // addCustomContent {
+        // Create a temporary dir
+        File contentFolder = new File(tmpDirectory);
         contentFolder.mkdir();
-        contentFolder.setReadable(true, false);
-        contentFolder.setWritable(true, false);
-        contentFolder.setExecutable(true, false);
-
+        contentFolder.deleteOnExit();
 
+        // And "hello world" HTTP file
         File indexFile = new File(contentFolder, "index.html");
-        indexFile.setReadable(true, false);
-        indexFile.setWritable(true, false);
-        indexFile.setExecutable(true, false);
-
+        indexFile.deleteOnExit();
         @Cleanup PrintStream printStream = new PrintStream(new FileOutputStream(indexFile));
-        printStream.println("<html><body>This worked</body></html>");
+        printStream.println("<html><body>Hello World!</body></html>");
+        // }
     }
 
     @Test
     public void testSimple() throws Exception {
+        // getFromNginxServer {
+        URL baseUrl = nginx.getBaseUrl("http", 80);
 
-        info("Base URL is " + nginx.getBaseUrl("http", 80));
+        assertThat("An HTTP GET from the Nginx server returns the index.html from the custom content directory",
+            responseFromNginx(baseUrl),
+            containsString("Hello World!")
+        );
+        // }
+    }
 
-        URLConnection urlConnection = nginx.getBaseUrl("http", 80).openConnection();
+    private static String responseFromNginx(URL baseUrl) throws IOException {
```

Comment:
This could be done inline but I moved to a separate method to make the example more readable.